src/objects.js

```diff
@@ -1,42 +1,49 @@
 const createPerson = (name, age) => {
-  // your code here
+  return {name, age} //Easy.
 };
 
 const getName = object => {
-  // your code here
+  return object.name; //Easy. Dot notation.
 };
 
 const getProperty = (property, object) => {
-  // your code here
+  return object[property]; // Bracket notation. Instead of using an index number to select an item, you are using the name associated with each member's value. Easy.
 };
 
 const hasProperty = (property, object) => {
-  // your code here
-};
+  return object.hasOwnProperty([property]); // Easy but had to use MDN. Returns true if the object has the specified property as own property.
+}; // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty
 
 const isOver65 = person => {
-  // your code here
+  return person.age > 65; //Easy. Just remember the object notation.
 };
 
 const getAges = people => {
-  // your code here
+  return people.map(person => person.age) //Struggled a bit. Kind of cheaty, as went back to arrays and used .map to creat new array.
 };
 
 const findByName = (name, people) => {
-  // your code here
+  return people.find (person => person.name === name) //Not too hard, still struggle a bit with the function parameters sometimes. Easyish.
 };
 
 const findHondas = cars => {
-  // your code here
+  return cars.filter (car => car.manufacturer === `Honda`); //Filter instead of find. Find returns first result, filter shows them all. Easy.
 };
 
 const averageAge = people => {
-  // your code here
-};
+  return people.reduce((a, b) => a + b.age, 0) / people.length; // Difficult. b.age -> had to declare .age. 
```

Comment:
This just needs a bit of familiarity to get your head around how reduce works, it gets easier with time!
In this case you are starting with an initial value of 0 (the second argument to `reduce`), and inside the callback (`(a, b) => a + b.age`) you are saying:
I have an array (`people`), and I want to start with the first two values in this array (we are calling them and b here) and add them together. Since the initial value you provided is 0, this is counted as the first value in the array and so is `a` here. `b` is the first object in the array, and since it is an object with a property we need to access, we do so using dot syntax (`b.age`). This total is then saved internally, and for the next iteration is used as the starting value `a`, and the next element in the array is then represented by `b`. The cycle continues until all the values in the array have been processed, and a single value has been arrived at

Using non generic names in place of `a` and `b` may help with understanding this in more detail, but dont worry. It'll come with time!